doc/developer/design/20220516_transactional_consistency.md

```diff
@@ -0,0 +1,416 @@
+
+Sinks are left out of this document and not considered for consistency.
+
+`TAIL` is left out of this document and not considered for consistency.
+
+`AS OF` allows clients to specify the timestamp of a query. This allows clients to circumvent our consistency
+guarantees, so they’re left out of this document.
+
+This document does not include Materialize initiated acknowledgements of upstream sources (e.g., the commit of an offset
+by a Kafka source, the acknowledgment of an LSN by a PostgresSQL source, etcetera). For the purposes of this document we
+will consider all of these events to have happened at the timestamp assigned to them by timely workers. This is not
+accurate for multiple reasons, some of which include:
+
+- Clock Skew: Each timely worker has their own clock that is used to timestamp events, that may be out of sync with
+  other clocks.
+- Event Time vs Process Time: Many events are timestamped with the time they arrive at Materialize not when they
+  actually happen. Network and other delays may make this timestamp inaccurate.
+- Reading from these sources almost always happen at a timestamp lower than what was just written to these source.
+- Etc.
+
+There will be a follow-up design document that goes into depth for these operations.
+
+## Description
+
+### Data Manipulation and Definition
+
+The main interface Materialize provides to clients for data manipulation and data definition is SQL. SQL provides
+transactions which allows a client to group multiple statements into a single atomic operation. Statements that are not
+explicitly inside a transaction should be considered to be implicitly inside a single statement transaction.
+
+#### Transactions
+
+Materialize supports two kinds of transactions, read-only and write-only transactions.
+
+##### Read Only Transactions
+
+Read only transactions allows a client to read from multiple objects at a single timestamp. When a client performs the
+first read of the transaction, a timestamp is selected for all subsequent reads and a read hold is taken on all objects
+in the same schema for that timestamp. A read hold prevents compaction from happening on an object up to that timestamp.
+A consequence of this is that read only transactions can only read from objects that are in the same schema and that
+existed at the time of the first read (due to implementation details on the read holds).
+
+There is an additional restriction that objects on different timelines cannot be read in the same transaction. Details
+on this are out of scope for this document but more details can be found
+here: [https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20210408_timelines.md](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20210408_timelines.md)
+
+##### Write Only Transactions
+
+Write only transaction allows a client to perform multiple writes atomically. They are currently very limited and only
```

Comment:
If I'm understanding your comment, maybe it would be best to just remove the "very limited" phrase and just write "They currently only allow writes to a single table"?

Reply:
sgtm